rfcs/0001-authentication/text.md

```diff
@@ -0,0 +1,383 @@
+
+- A request is sent to the to the “/logout” endpoint in cellxgene
+- cellxgene redirects back to the current page, with a directive to delete the cookie containing the JWT.
+
+#### Application Endpoints
+
+To support the Auth0 authentication, following endpoints are needed by cellxgene. These endpoints were referenced in the previous sections, and defined here in more detail.
+
+Note: If authentication is not enabled, then these endpoints will not be present (e.g. they would all return 404 errors).
+
+- /login : this endpoint is used to initiate the login process. The client passes in the
+  location to the current dataset in the request. The server then initiated the authentication
+  request by redirecting the browser to the Auth0 universal login page. The server passes in the
+  location of the dataset as a state variable in the request, so the browser can be redirected
+  there after login. This is step 1 in the diagram. /login is requested when the user clicks on
+  the “login” button.
+- /logout : this endpoint is used to clear the authentication credentials. The cookie
+  containing the ID token is removed. /logout is requested when the user clicks on the “logout” button.
+- /oauth2/callback: this endpoint is accessed during step 3 of the diagram. Auth0 redirects
+  the user to this endpoint, and the request contains the authentication code and the state
+  parameter sent in from the /login endpoint. The cellxgene server then exchanges the
+  authentication code for an ID Token. Once complete, the cellxgene server redirects the client
+  back to the dataset, and directs the browser to store the ID Token in a cookie.
+
+### Authentication in Data Portal
+
+When a user visits the Data Portal site, we determine if the user is authenticated or not by
+querying a backend endpoint (e.g. api.cellxgene.cziscience.com/corpora/is-authenticated). This
+can return back the username if authenticated, or a null response if not authenticated.
+
+If authenticated, a Logout button may appear near the top of the page.
+
+If unauthenticated, a Login button may appear near the top of the page, with the name of the user.
+
+In a previous version of this design doc, the authentication strategy for Data Portal was to use
+the [Authentication Code Flow with Proof Key for Code Exchange (PKCE)](https://auth0.com/docs/flows/concepts/auth-code-pkce),
+implemented in the client side. This has now been changed to
+using the Authentication Code flow in the backend (same as cellxgene). This approach is
+simpler with respects to the single sign on strategy. It avoids a step where the client
+needs to access the server to set the cookie containing the tokens. It is also more secure.
+
+The Data Portal front end requires backend support for the following endpoints
+
+- Login: start the authentication flow, set the cookie when complete
+- Logout: logout, remove the cookie
+- Is-authenticated: check if authenticated, and return user information
```

Comment:
Sorry, I can't remember if this was built or not, but can you replace with the actual endpoint name if it was?